### PR TITLE
Set assistant prompt from URL

### PR DESCRIPTION
**Pull request by nsarrazin**
### Assistant view
image
### Settings edit/new
image

This uses the same `parseWeb` method we use in the web search, but instead of chunking and generating embeddings we just add the resulting text to the system prompt.

### CODE REVIEW

File: src/lib/components/AssistantSettings.svelte [hunk 1]
```diff
@@ -119,18 +121,29 @@
...
 			formData.set("ragAllowAll", "false");
 			formData.set("ragLinkList", "");
+			formData.set("ragPrepromptUrl", "");
+		} else if (ragMode === "preprompt") {
+			formData.set("ragAllowAll", "false");
+			formData.set("ragLinkList", "");
+			formData.set("ragDomainList", "");
+			formData.set(
+				"preprompt",
+				"The system prompt input is disabled when using a system prompt URL."
```
Comment (gary149) [inline src/lib/components/AssistantSettings.svelte:144]: We could just refactor later no? I don't get why we need this list this should be handled natively by the <form>.

File: src/routes/conversation/+server.ts [hunk 2]
```diff
@@ -51,11 +53,23 @@ export const POST: RequestHandler = async ({ locals, request }) => {
 	});
 
 	if (assistant) {
-		values.preprompt = assistant.preprompt;
-	} else {
-		values.preprompt ??= model?.preprompt ?? "";
+		console.log(assistant?.rag?.prepromptUrl);
```
Comment (gary149) [inline src/routes/conversation/+server.ts:56]:
```suggestion

```

File: src/routes/conversation/+server.ts [hunk 3]
```diff
@@ -51,11 +53,22 @@ export const POST: RequestHandler = async ({ locals, request }) => {
 	});
 
 	if (assistant) {
-		values.preprompt = assistant.preprompt;
-	} else {
-		values.preprompt ??= model?.preprompt ?? "";
+		// if the assistant has a preprompt Url and it's not local, try to fetch it
+		if (assistant.rag?.prepromptUrl && !(await isURLLocal(new URL(assistant.rag.prepromptUrl)))) {
+			try {
+				const content = await parseWeb(assistant.rag.prepromptUrl);
```
Comment (gary149) [inline src/routes/conversation/+server.ts:59]: I would not parse like for web content.

**Comment by secondtruth** [comment]
I think, this will be a very useful feature. 👍

But I would suggest to make the internet access features composable, so that web search, prompts from URLs, etc. can be used together if desired. In that case, the settings UI could be structured this way:

> **Internet access**
> - [Checkbox] **Web search** – _expands the options:_
>   - (Radio) **Everywhere**
>   - (Radio) **Specific Domains**
>   - (Radio) **Specific Links**
> - [Checkbox] **System Prompt URL**

---

**Related to this feature:**
- #611 (Part "**Prompts from external APIs**")

**Comment by nsarrazin** [comment]
Closing in favour of #949

**Pull request closed without merging**